docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,405 @@
+
+##### MeshRateLimit
+
+###### Old ExternalService behaviour
+
+Both RateLimit and MeshRateLimit are configured on the egress.
+
+###### New behaviour
+
+If on sidecar:
+- limit is multiplied by the number of instances (which is bad)
+- less traffic in the mesh (traffic does )
+
+
+##### MeshRetry
+
+###### Old ExternalService behaviour
+
+Both Retry and MeshRetry are configured on the sidecar
+
+###### New behaviour
+
+If on egress:
+- will behave the same as on sidecar
+- we definitely shouldn't configure it both on sidecar and egress, that way we'll have squared retries
+
+##### MeshTimeout
+
+###### Old ExternalService behaviour
+
+Both Timeout and MeshTimeout are configured on the sidecar.
+
+###### New behaviour
+
+If on egress:
+- we would have to match the timeouts from the sidecar not to have a clash (that server times out before client)
+- I don't see any advantages of having this on egress but nothing rules it out
+
+##### MeshTrace
+
+###### Old ExternalService behaviour
+
+Both TrafficTrace and MeshTrace do not support egress.
+
+###### New behaviour
+
+I think it makes sense both on sidecar and egress.
+If we're troubleshooting something it would be great to see spans from sidecar going to ExternalService.
+
+##### MeshTrafficPermission
+
+###### Old ExternalService behaviour
+
+Both MeshTrafficPermission and TrafficPermission only works with egress.
+
+###### New behaviour
+
+It is already implemented on egress for MeshTrafficPermission.
+
+If on the sidecar:
+- could you circumvent this limitation by trying to directly call through egress?
+
+##### MeshLoadBalancingStrategy
+
+###### Old ExternalService behaviour
+
+There is no old behaviour because there was no concept of Endpoints in External Service
+
+###### New behaviour
+
+Only the `loadBalancer` field makes sense in this regard.
+
+If on sidecar:
+- if you don't need/want egress it still makes sense to pick load balancing mechanism
+
+If on egress:
+- makes the same sense on as on sidecar
+
+##### MeshTCPRoute and MeshHTTPRoute
+
+###### Old ExternalService behaviour
+
+It worked for both egress and sidecar.
+
+###### New behaviour
+
+I don't see why we would change that behaviour.
+
+#### For outbound policies on the sidecar, for inbound policies on egress
+
+It naturally fits our targetRef mechanism.
+It does not make sense to apply inbound policies on the sidecar since there is no inbound for it.
+Egress plays a role of a "sidecar" that is inside the mesh, and we can target it's inbound.
+Configuring outbound policies on the sidecar avoids the problem with different clients influencing each other (see [MeshCircuitBreaker](#meshcircuitbreaker)).
+
+##### How to configure it
+
+We decide to implicitly configure sidecar and egress based on the policy.
+
+###### `from` only policy
+
+Policies:
+- MeshTrafficPermission
+- MeshRateLimit
+- MeshGlobalRateLimit
+- MeshFaultInjection
+
+```yaml
+kind: MeshTrafficPermission
+spec:
+  targetRef:
+    kind: MeshExternalService
```

Comment:
Shouldn't `spec.targetRef` only target the proxies the config affects? How does this fit in with all the ideas about "consistency" and https://github.com/kumahq/kuma/pull/10858?
`MeshExternalService` is just a destination, why don't we allow `spec.targetRef.kind: MeshService`?
We changed this already and it was apparently a big mistake, what exactly does `spec.targetRef` mean?

Reply:
How do we integrate this with https://kuma.io/docs/2.8.x/policies/targetref/#understanding-targetref-policies for example

Reply:
Once we agreed to select dataplane but it seems we don't have a way of targeting ZoneEgress and a specific part of it which is required here. It requires a design but it's out of the scope of this MADR